lib/ansible/modules/system/gconftool2.py

```diff
@@ -36,146 +38,233 @@
+        :type module: AnsibleModule
+
+        :param direct: Specify whether the GConf repository should be accessed directly or through gconfd-2 daemon.
+                       If enabled, valid config_source must be passed in as well.
+        :type direct: bool
+
+        :param config_source: Configuratrion source to use in case direct access has been requested.
+        :type config_source: str
+
+        :param check_mode: Specify whether to only check if a change should be made or if to actually make a change.
+        :type check_mode: bool
+        """
+
+        self.module = module
+        self.base_command = ["gconftool-2"]
+        self.check_mode = check_mode
+
+        if direct:
+            self.base_command += ["--direct", "--config-source", config_source]
+
+    def get(self, key):
+        """
+        Retrieves current value associated with the key.
+
+        If an error occurs, a call will be made to AnsibleModule.fail_json.
+
+        :returns: string -- Value assigned to the provided key. If the key does not exist (i.e. no value is assigned), returns None.
+        """
+
+        # Set-up command to run.
+        command = list(self.base_command)
+        command += ["--get", key]
+
+        # Run the command and fetch standard return code, stdout, and stderr.
+        rc, out, err = self.module.run_command(command)
+
+        # Since the tool does not produce proper error return codes, we need to
+        # process the stderr output to determine if there are issues. This could
+        # be prone to localisation issues!
+        if len(err) > 0 and not err.startswith('No value set'):
+            self.module.fail_json(msg='gconftool-2 failed while reading the value from GConf with error: %s' % (str(err)))
+
+        # Standard output will contain the value, or message that no value is
+        # set at all.
+        if err.startswith('No value set'):
+            value = None
+        else:
+            value = out.rstrip("\n")
+
+        return value
+
+    def set(self, key, value_type, value):
+        """
+        Sets the value for specified key.
+
+        If an error occurs, a call will be made to AnsibleModule.fail_json.
+
+        :param key: GConf preference key to set. Should be a full path.
+        :type key: str
+
+        :param value_type: GConf value type. See gconftool-2 for supported values.
+        :type value_type: str
+
+        :param value: Value to set for the specified GConf preference key.
+        :type value: str
+
+        :returns: bool -- True if a change was made, False if no change was required.
+        """
+
+        # If no change is needed (or won't be done due to check_mode), notify
+        # caller straight away.
+        if value == self.get(key):
+            return False
+        elif self.check_mode:
+            return True
+
+        # Set-up command to run.
+        command = list(self.base_command)
```

Comment:
The intention is to create copy of `base_command` - which I keep intact since both `set` and `unset` use it as basis for constructing relevant commands, and both `set` and `unset` will call the `read` method at least once. .